library/std/src/sys/theseus/thread_local_key.rs

```diff
@@ -1,3 +1,6 @@
+//! Theseus has support for TLS sections and so these functions should not be
+//! used.
+
```

Comment:
true, but wouldn't we need to re-export our implementation details in `thread_local_macro` such that this crate can use them?

Reply:
No. `thread_local_macro` is just a copy of the `std` file; they both use `thread_local` LLVM attribute. https://github.com/rust-lang/rust/blob/4891d57f7aab37b5d6a84f2901c0bb8903111d53/library/std/src/thread/local.rs#L204-L256

I had to get thread locals working for `println!` in the [`test_std`](https://github.com/theseus-os/Theseus/blob/std_dep/applications/test_std/src/lib.rs) application.

Reply:
so all we need to do here is to set the thread local item to true in Theseu's target spec, yes?

Reply:
yep